cfn-resources/third-party-integration/cmd/resource/resource.go

```diff
@@ -150,132 +129,109 @@ func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler
 
 	_, _ = log.Debugf("Update() currentModel:%+v", currentModel)
 
-	// Validation
 	if modelValidation := validateModel(RequiredFields, currentModel); modelValidation != nil {
 		return *modelValidation, nil
 	}
 
-	// Create atlas client
-	if currentModel.Profile == nil || *currentModel.Profile == "" {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
-	}
-
-	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	client, peErr := util.NewAtlasClient(&req, currentModel.Profile)
 	if peErr != nil {
 		return *peErr, nil
 	}
 
 	ProjectID := currentModel.ProjectId
 	IntegrationType := currentModel.Type
 
-	integration, res, err := client.Integrations.Get(context.Background(), *ProjectID, *IntegrationType)
+	integration, res, err := client.AtlasV2.ThirdPartyIntegrationsApi.GetThirdPartyIntegration(context.Background(), *ProjectID, *IntegrationType).Execute()
 	if err != nil {
-		_, _ = log.Debugf("Update - error: %+v", err)
-		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+		return progressevent.GetFailedEventByResponse(err.Error(), res), nil
 	}
 
-	// check for changed attributes per type
-
 	updateIntegrationFromSchema(currentModel, integration)
-
-	integrations, res, err := client.Integrations.Replace(context.Background(), *ProjectID, *IntegrationType, integration)
-
+	integrations, res, err := client.AtlasV2.ThirdPartyIntegrationsApi.UpdateThirdPartyIntegration(context.Background(), *IntegrationType, *ProjectID, integration).Execute()
 	if err != nil {
-		_, _ = log.Debugf("Update - error: %+v", err)
-		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+		return progressevent.GetFailedEventByResponse(err.Error(), res), nil
 	}
-	_, _ = log.Debugf("Atlas Client %v", client)
 
-	// Response
 	return handler.ProgressEvent{
 		OperationStatus: handler.Success,
-		ResourceModel:   integrationToModel(*currentModel, integrations.Results[0]),
+		ResourceModel:   integrationToModel(*currentModel, &integrations.Results[0]),
 	}, nil
 }
 
-func updateIntegrationFromSchema(currentModel *Model, integration *mongodbatlas.ThirdPartyIntegration) {
-	if currentModel.Url != nil && *currentModel.Url != integration.URL {
-		integration.URL = *currentModel.Url
+func updateIntegrationFromSchema(currentModel *Model, integration *admin.ThridPartyIntegration) {
+	if currentModel.Url != nil && *currentModel.Url != *integration.Url {
```

Comment:
I think currentModel.Url is *string, do we want to assign if *currentModel.Url == "", the if clause will be true. 
same for all conditions here.

in case we don't want to assign if it is empty, we can do something like this: 
if util.IsStringPresent(currentModel.Url) && !util.AreStringPtrEqual(currentModel.Url, integration.Url) {
although to be honest i don't think it makes any harm to assign even if it's the same, we could just have: 

if  util.IsStringPresent(currentModel.Url) {
	integration.Url = currentModel.Url
}

Reply:
I see you're doing my last idea in fucntion modelToIntegration, why not here?

Reply:
Both options are okay with me, mainly want to avoid the `*integration.Url` as it has the risk of nil pointer dereference. (Applies to other optional properties below in this function)

Reply:
I agree that they behave similarly, although I find easier to understand and cleaner: if util.IsStringPresent(currentModel.Url) {

Reply:
> I see you're doing my last idea in fucntion modelToIntegration, why not here?
the only reason is that `modelToIntegration` was only checking for empty string whereas `updateIntegrationFromSchema` check that the current model was different from the input

Reply:
ok, but I don't understand yet why we use !util.AreStringPtrEqual in one method and not in the other, what's the different logic we want to achieve?

Reply:
`modelToIntegration` is used for the CREATE so we know there is no resource in Atlas. `updateIntegrationFromSchema` is used in the UPDATE, so we want to send in the body request only the fields that have been changed in the currentModel. The main point here is that we have fields such as API tokens that we don't want to send again into the API call if not really needed.